docs/UserGuide.md

```diff
@@ -45,54 +45,108 @@ By : `Team SE-EDU`  &nbsp;&nbsp;&nbsp;&nbsp; Since: `Jun 2016`  &nbsp;&nbsp;&nbs
 
 Format: `help`
 
+> A list of commands available will be shown
 > Help is also shown if you enter an incorrect command e.g. `abcd`
 
-### 2.2. Adding a person: `add`
+Format: `help COMMAND`
 
-Adds a person to the address book<br>
-Format: `add NAME p/PHONE_NUMBER e/EMAIL a/ADDRESS [t/TAG]...`
+> Different type of commands relating to COMMAND is shown 
+> e.g. `help help`
 
-> Persons can have any number of tags (including 0)
+### 2.2.1. Adding a task: `add`
 
-Examples:
+Adds a task to the task manager
+Format: `add DESCRIPTION d/DATE t/TIME f/FREQUENCY bt/TIMERANGE p/PRIORITY [TAG]`
+
+> * DESCRIPTION represent the task (complusory)
+> * d/ represents the deadline date*
+> * t/ represents the deadline time
+> * f/ represents the type to repeat, e.g. w / m / y
+> * bt/ represents the timeslot to be blocked, must be given in a range
+> * p/ represents the priority**
+> * [TAG] represents the tag***
+
+Note
+*If DATE is not specified, today's date will be selected
+**Tasks can have any level of priority (including 0)
+***Tasks can have any number of tags (including 0)
+
+### 2.2.2. Adding tags to existing tasks: `addtag`
+
+Adds tag to existing tasks
+Format: `addtag INDEX [TAG]...`
+> * Adds tag to task at the specified INDEX
+    The index refers to the index number shown in the last task listing.<br>
+    The index **must be a positive integer** 1, 2, 3, ...
+> * More tags will be added to the task
+
+
+### 2.3.1. Listing all tasks : `list`
+
+Shows a list of all tasks in the task manager<br>
+Format: `list d/DATE d/DATE p/PRIORITY [TAG]...`
+> * All fields are not compulsory
+> * d/ would show the tasks due by specified deadline*
+> If two DATE are given, tasks due within the period would be shown
+> * p/ would show the tasks of that priority*
+> * [TAG] would show tasks under the tag, allow for multiple
 
-* `add John Doe p/98765432 e/johnd@gmail.com a/John street, block 123, #01-01`
-* `add Betsy Crowe t/friend e/betsycrowe@gmail.com a/Newgate Prison p/1234567 t/criminal`
+Note
+*If `d/` or `p/` is given, but no specified DATE or PRIORITY, the list would be sorted by the given command
 
-### 2.3. Listing all persons : `list`
+### 2.3.2. Listing all completed tasks : `listcom`
 
-Shows a list of all persons in the address book.<br>
-Format: `list`
+Shows a list of all completed tasks in the task manager<br>
+Format: `listcom`
 
-### 2.4. Editing a person : `edit`
+### 2.3.2. Listing all tags : `listtag`
```

Comment:
Consider if you can combine all the variants of listing under the `list` command?  What if I want to list completed tasks within a certain period?  How about listing uncompleted tasks?